### ISSUE TITLE
Squared orientation only.

### ISSUE THREAD

**Issue by kilinkis** [stevenschobert/instafeed.js#372]
Hello guys

I'd like to get squared images only (not images with white border, nor landscape or portrait).
is this possible? how?

thanks!

**Comment by Tardis1234** [comment]
Take a look at [this section](https://github.com/stevenschobert/instafeed.js#portrait-and-landscape-photos) of the docs.

**Comment by ongerit** [comment]
@kilinkis- Please close this issue.

**Comment by christinejoy** [comment]
Hi- I am having this same issue, I would like my instafeed images all square... I have read the sec in docs and have tried to apply the default to the resolution without success. Any suggestions?

http://smittentestblogcorinneperry.blogspot.com

Thanks!

**Comment by alfakappa** [comment]
the funny thing is; if I request the orientation, it is returning "square" for all images, although I explicitly turned it on in my app.

Any others having the same problem?

**Comment by randomobject** [comment]
Hey @alfakappa  I'm having similar issue. Seems that turning on 'Non square media' in the migration section of the instagram developer settings has not effect. I'm using standard_resolution and still getting bars wit the 'Non square media' checked. 

Have you managed to resolve this?

**Comment by alfakappa** [comment]
Unfortunately not. Which is really annoying, cause I really need higher res pictures for my application (which is a prototype at the moment). Hope Plugin-owner will look into this on short notice...

**Comment by paulasolanoy** [comment]
Hi guys! I'm also still getting the white borders even thought I had already checked off the "Non square media" on my Instagram account. Anyone know a work around or solution?

Thanks,
Paula

**Comment by alfakappa** [comment]
@ongerit It should nog be closed, since issue is still alive... Do you know if Plugin-owner has a solution?

**Comment by stevenschobert** [comment]
Sorry everyone, I've been busy with some other projects.

Since the API doesn't provide urls for square versions of the images, I can't build in any options to automatically do this for you.

The best solution is going to be to use CSS + the templating options to control the size of the images.

The basic idea is to use create a fixed size element, with the dimensions set to a square ratio, and then set the Instagram image as the background image of the element. This will basically crop any overflowing parts of the image by hiding them.

Here is a basic example on StackOverflow you can use as a guide:

http://stackoverflow.com/questions/11552380/how-to-automatically-crop-and-center-an-image

**Comment by alfakappa** [comment]
Yeah... that is something you could do, however... it is returning the same class for all images (being "square") even when it is not a square image.

So that is actually the problem (for me)

**Comment by stevenschobert** [comment]
@alfakappa Are you saying the `{{orientation}}` template helper setting the value as square? Do you have an example I can look at?

**Comment by alfakappa** [comment]
@stevenschobert yep, that's exactly what I mean. it is returning "square" for all images. I do not have public example available, but here is my code:

var feed = new Instafeed({
                get: 'user',
                userId: '*******',
                clientId: '********',
                limit: 9,
                sortBy: 'random',
                resolution: 'standard_resolution',
                template: '<div class="instaTile {{orientation}}"><img src={{image}} /></div>',
            });
            feed.run();

            setInterval(function() {
                //empty feed and reset counter
                $( "#instafeed" ).empty();
                count = 0;
                //run feed again
                feed.run();
            }, 20000);

(sorry code block is not working probably with html in script)

**Comment by randomobject** [comment]
@stevenschobert +1 for @alfakappa message above as we are having the same problem.

my experience is that the images are all returning square. i.e. white borders on landscape and portrait images. when I use the helper `{{orientation}}` it give the class `instagram-square`, which of course they are as they have white borders. they option in instagram to early migrate to the new orientation has no effect. Turned on or off, landscape and portrait images are all returned with white borders / square.

hope this helps to clarify?

**Comment by stevenschobert** [comment]
@alfakappa Ahhh okay. At first I thought you were saying the images _were_ different dimensions, but just the helper was returning the wrong value, which would be a different problem.

Whether or not the API request returns images as bordered squares or their original sizes is controlled by the API client settings, and consequently your access token. Have you double checked that you have the correct settings in your API client, and that your access token was generated **after** you updated your settings?

**Comment by paulasolanoy** [comment]
I generated my access token after changing all the API settings and it was still returning the white borders. Hmm...

**Comment by randomobject** [comment]
Yep, still no dice with me too. I've generated my access token after API settings change. Created a new one. I've even now deleted the client altogether, which oddly has also had no effect - feed still works (with the white borders that is).

**Comment by stevenschobert** [comment]
How did you generate the access token?

**Comment by randomobject** [comment]
@stevenschobert thanks for the prompt. I figured it out! - I originally registered a new client in **my** instagram developer account, so the `clientId` and `accessToken`, which I generated using the the step outlined by instagram [here](https://www.instagram.com/developer/authentication/) and a simpler version [here](https://instagram.com/oauth/authorize/?client_id=[CLIENT_ID_HERE]&redirect_uri=http://localhost&response_type=token) came from the 'client' I had registered - when I follow the steps to create the access token I was given a token for **my**  instagram account.

The site I am implementing this on, is for a physical client / another person so I had used their `userId` - which conflicted with the `clientId` (from my account/registered client) and `accessToken` (again generated from my account/registered client).

Basically, what I did to fix this was: log into my **clients** instagram account, created a developer account, registered a new client, followed the access token steps from above, using details from this client and it all works perfectly. Make sense? 

Hope it helps.

**Comment by stevenschobert** [comment]
Yep that makes perfect sense.

Access tokens are tied to the client application that issued them. So in order to get the new non-square images, you need to be using a token that was issued by a client application with that setting enabled.

Glad you were able to find the answer!

**Comment by randomobject** [comment]
thanks for taking time out to help.
